tools/maven-plugin/src/main/java/io/smallrye/openapi/mavenplugin/MavenDependencyIndexCreator.java

```diff
@@ -0,0 +1,180 @@
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.component.annotations.Component;
+import org.codehaus.plexus.component.annotations.Requirement;
+import org.codehaus.plexus.logging.Logger;
+import org.jboss.jandex.CompositeIndex;
+import org.jboss.jandex.Index;
+import org.jboss.jandex.IndexView;
+import org.jboss.jandex.Indexer;
+import org.jboss.jandex.JarIndexer;
+import org.jboss.jandex.Result;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+
+@Component(role = MavenDependencyIndexCreator.class, instantiationStrategy = "singleton")
+public class MavenDependencyIndexCreator {
+
+    private final Cache<String, IndexView> indexCache = CacheBuilder.newBuilder().build();
+
+    private final Set<String> ignoredArtifacts = new HashSet<>();
+
+    @Requirement
+    private Logger logger;
+
+    public MavenDependencyIndexCreator() {
+        ignoredArtifacts.add("org.graalvm.sdk:graal-sdk");
+        ignoredArtifacts.add("org.yaml:snakeyaml");
+        ignoredArtifacts.add("org.wildfly.common:wildfly-common");
+        ignoredArtifacts.add("com.fasterxml.jackson.core:jackson-core");
+        ignoredArtifacts.add("com.fasterxml.jackson.core:jackson-databind");
+        ignoredArtifacts.add("io.quarkus:quarkus-vertx-http");
+        ignoredArtifacts.add("io.smallrye:smallrye-open-api-core");
+        ignoredArtifacts.add("io.smallrye.reactive:smallrye-mutiny-vertx-core");
+        ignoredArtifacts.add("commons-io:commons-io");
+        ignoredArtifacts.add("io.smallrye.reactive:mutiny");
+        ignoredArtifacts.add("org.jboss.narayana.jta:narayana-jta");
+        ignoredArtifacts.add("org.glassfish.jaxb:jaxb-runtime");
+        ignoredArtifacts.add("com.github.ben-manes.caffeine:caffeine");
+        ignoredArtifacts.add("org.hibernate.validator:hibernate-validator");
+        ignoredArtifacts.add("io.smallrye.config:smallrye-config-core");
+        ignoredArtifacts.add("com.thoughtworks.xstream:xstream");
+        ignoredArtifacts.add("com.github.javaparser:javaparser-core");
+        ignoredArtifacts.add("org.jboss:jandex");
+
+        ignoredArtifacts.add("antlr");
+        ignoredArtifacts.add("io.netty");
+        ignoredArtifacts.add("org.drools");
+        ignoredArtifacts.add("net.bytebuddy");
+        ignoredArtifacts.add("io.vertx");
+        ignoredArtifacts.add("org.hibernate");
+        ignoredArtifacts.add("org.kie");
+        ignoredArtifacts.add("org.postgresql");
+        ignoredArtifacts.add("org.apache.httpcomponents");
```

Comment:
Not quit sure about having to maintain a list of blacklisted artifacts.
It does bring about 1s though, which is justifiable (imo) for a larger multi-module projects.

WDYT?